src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -140,59 +142,37 @@
     List<ExceptionCaseWorker> upExceptionCaseWorkers;
 
 
-    @Override
-    public List<CaseWorkerProfile> processCaseWorkerProfiles(List<CaseWorkersProfileCreationRequest> cwRequests) {
-        List<CaseWorkerProfile> newCaseWorkerProfiles = new ArrayList<>();
-        List<CaseWorkerProfile> updateCaseWorkerProfiles = new ArrayList<>();
-        Map<String, CaseWorkersProfileCreationRequest> requestMap = new HashMap<>();
-        List<CaseWorkerProfile> processedCwProfiles = new ArrayList<>();
+    public List<CaseWorkerProfile> processCaseWorkerProfiles(List<CaseWorkersProfileCreationRequest> cwUiRequests) {
+
+        List<CaseWorkerProfile> processedCwProfiles;
+        List<CaseWorkerProfile> newCaseWorkerProfiles;
+        Map<String, CaseWorkersProfileCreationRequest> emailToRequestMap = new HashMap<>();
+
+        //create map for input request to email
+        for (CaseWorkersProfileCreationRequest request : cwUiRequests) {
+            emailToRequestMap.put(request.getEmailId().toLowerCase(), request);
+        }
+
         try {
+            // get all existing profiles from db (used IN clause)
+            List<CaseWorkerProfile> cwDbProfiles = caseWorkerProfileRepo.findByEmailIdIn(emailToRequestMap.keySet());
```

Comment:
emailToRequestMap if empty then no point to make db call

Reply:
This will only happen if blank excel file is uploaded and already have validation for it, so i dont think emailToRequestMap will be empty anytime.